core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -186,7 +187,7 @@ public final class MetadataManager
     @VisibleForTesting
     public static final int MAX_TABLE_REDIRECTIONS = 10;
 
-    private final GlobalFunctionCatalog functions;
+    private final FunctionProvider functions;
```

Comment:
It's not some "function provider". Maybe perhaps we need FunctionManager interface or something like that to denote the global function catalog?

Reply:
Separate interface CatalogFunctionManager introduced